src/main/java/org/springframework/data/jpa/repository/support/SimpleJpaRepository.java

```diff
@@ -620,6 +621,20 @@ public long count(@Nullable Specification<T> spec) {
 		return result;
 	}
 
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.jpa.repository.JpaRepository#saveAllAndFlush(java.lang.Iterable)
+	 */
+	@Transactional
+	@Override
+	public <S extends T> List<S> saveAllAndFlush(Iterable<S> entities) {
+
+		List<S> result = saveAll(entities);
+		flush();
```

Comment:
Rather than call `saveAll` and `flush` directly here it might be preferable to copy their contents into this method just in case, or document that you are calling those methods in the JavaDoc. That way any users that extend the public class SimpleJpaRepository and override saveAll or flush will know that their overridden methods can be called by this method. Might be best to discuss with maintainers how best to approach this here.

Reply:
Good point! I took a look at the implementations of the method that performs this action for a single entity:

```
/**
* Saves an entity and flushes changes instantly.
*
* @param entity
* @return the saved entity
*/
<S extends T> S saveAndFlush(S entity);
```

```
/*
* (non-Javadoc)
* @see org.springframework.data.jpa.repository.JpaRepository#saveAndFlush(java.lang.Object)
*/
@Transactional
@Override
public <S extends T> S saveAndFlush(S entity) {

S result = save(entity);
flush();

return result;
}
```

Arguably I would say adding a comment in the javadocs would indeed be ideal, but the current implementation adheres to the standard that has been used previously. Shall I make a PR to the main repository and include a comment with this remark?

Reply:
yeah sure! Think this is ready for further feedback there; can't hurt to mention this. :)